cmd/zc_traverser_blob.go

```diff
@@ -142,11 +142,14 @@ func (t *blobTraverser) traverse(preprocessor objectMorpher, processor objectPro
 		searchPrefix += common.AZCOPY_PATH_SEPARATOR_STRING
 	}
 
+	// as a performance optimization, get an extra prefix to do pre-filtering. It's typically the start portion of a blob name.
+	extraSearchPrefix := filterSet(filters).GetEnumerationPreFilter()
+
 	for marker := (azblob.Marker{}); marker.NotDone(); {
 		// look for all blobs that start with the prefix
 		// TODO optimize for the case where recursive is off
 		listBlob, err := containerURL.ListBlobsFlatSegment(t.ctx, marker,
-			azblob.ListBlobsSegmentOptions{Prefix: searchPrefix, Details: azblob.BlobListingDetails{Metadata: true}})
+			azblob.ListBlobsSegmentOptions{Prefix: searchPrefix + extraSearchPrefix, Details: azblob.BlobListingDetails{Metadata: true}})
```

Comment:
This... is actually overall a bug, and I don't think we can get past it.

Unless we're in a virtual folder with files like this, a filter of a* can cause a return of no results.

Rather than, as a user might expect, returning results of all files in sub-virtual-directories containing this pattern of a*.

Reply:
I think the best path forward _may_ be to just dynamically decrease the size of our estimated job part, as I'd suggested on teams, on second thought.

Reply:
Oh, yes, well spotted. It is indeed a bug.

Reply:
Tweaking job part size solves a different problem (slow to start first transfer).  But in this PR we're trying to find a way to efficiently select only a subset of what is in the container.